middleware.ts

```diff
@@ -0,0 +1,12 @@
+import { createMiddlewareClient } from '@supabase/auth-helpers-nextjs'
+import { NextResponse } from 'next/server'
+
+import type { NextRequest } from 'next/server'
+import type { Database } from '@/types_db'
+
+export async function middleware(req: NextRequest) {
+  const res = NextResponse.next()
+  const supabase = createMiddlewareClient<Database>({ req, res })
+  await supabase.auth.getSession()
+  return res
```

Comment:
@thorwebdev could you help explain what this does? I tried removing it and things seem to work, though likely I am misunderstanding something?

Reply:
cc @dalkommatt perhaps you could explain?

Reply:
@jparismorgan you should look at the supabase docs for nextjs app directory integration